In [1]:
import pandas as pd
import datetime
import holidays

In [2]:
dati = pd.read_csv(r'Dati_PUN_17_19.csv', parse_dates = ['Data'])

In [3]:
dati['Giorno'] = dati['Data'].apply(lambda x: x.date().strftime('%A'))

In [4]:
iatly_holidays = holidays.Italy(years = [2017, 2018, 2019])

In [5]:
def get_holiday(row):
    row = row.date()
    iatly_holidays = holidays.Italy(years = [2017, 2018, 2019])
    if row in iatly_holidays:
        return(1)
    else:
        return(0)    

In [6]:
dati['Giorno_Vacanza'] = dati['Data'].apply(get_holiday)

In [7]:
dati.head()

Data    PUN  Giorno  Giorno_Vacanza
0 2017-01-01 00:00:00  53.30  Sunday               1
1 2017-01-01 01:00:00  52.00  Sunday               1
2 2017-01-01 02:00:00  51.00  Sunday               1
3 2017-01-01 03:00:00  47.27  Sunday               1
4 2017-01-01 04:00:00  45.49  Sunday               1

In [8]:
dati['Giorno_Vacanza'].value_counts()

0    25416
1      864
Name: Giorno_Vacanza, dtype: int64

Corretto poichè ci sonp 12 giorni festivi spalamati su 24 ore e 3 anni. Quindi 12 * 24 * 3 = 864